src/realm/mixed.cpp

```diff
@@ -448,6 +444,99 @@ util::Optional<UUID> Mixed::get<util::Optional<UUID>>() const noexcept
     return get<UUID>();
 }
 
+static DataType get_common_type(DataType t1, DataType t2)
+{
+    // FIXME: only a few types supported
+    if (t1 == t2)
+        return t1;
+    if (t1 == type_Decimal || t2 == type_Decimal)
+        return type_Decimal;
+    if (t1 == type_Double || t2 == type_Double)
+        return type_Double;
+    if (t1 == type_Float || t2 == type_Float)
+        return type_Float;
+    return type_Int;
+}
+
+Mixed Mixed::operator+(const Mixed& rhs) const
+{
+    if (!is_null() && !rhs.is_null()) {
+        auto common_type = get_common_type(get_type(), rhs.get_type());
+        switch (common_type) {
+            case type_Int:
+                return export_to_type<Int>() + rhs.export_to_type<Int>();
+            case type_Float:
+                return export_to_type<float>() + rhs.export_to_type<float>();
+            case type_Double:
+                return export_to_type<double>() + rhs.export_to_type<double>();
+            case type_Decimal:
+                return export_to_type<Decimal128>() + rhs.export_to_type<Decimal128>();
```

Comment:
Updated `export_to_type`. I am not sure what you mean by the last sentence. `export_to_type` should only be called for numeric values.

Reply:
Oh, I thought that this code would assert if the arithmetic operates on two mixed properties when both values are strings, but I can see now that the type is checked so that doesn't happen.